README.md

```diff
@@ -1,2 +1,208 @@
-# phplist-plugin-simplesaml
+# phpList simpleSAML Plugin
+
 SimpleSaml plugin for phpList
+
+## Plugin Setup
+
+After `cd`-ing into the configured phpList plugin directory:
+
+- `git clone https://github.com/phpList/phplist-plugin-simplesaml.git`
+- `composer install`
+- `cp .env.example. .env`
+- Ensure `SIMPLESAMLPHP_INSTALLTATION_PATH` in `.env` is the correct relative path installation of `simplesamlphp`
+
+# SimpleSAMLPHP Setup
+
+- `cd phplist-plugin-simplesaml/plugins/authsaml/simplesamlphp`
+- `composer install`
+- `npm install`
+- `npm run build`
+
+NB : `simplesamlphp` **MUST BE CONFIGURED ON SAME DOMAIN** as your phplist installation. Hence the right place to do the clone is in `PATH_TO_PHPLIST_INSTALLATION/public_html`
+
+You might be required to manually create a `log` directory with write permissions granted, so: `mkdir log && sudo chmod -R a+rwx log`
+
+## Configuration
+
+In `PATH_TO_PHPLIST_INSTALLATION/public_html/simplesamlphp` the following directories should be present.
+
+- `config`
+- `metadata`
+
+If not, you want to:
+
+- `cd PATH_TO_PHPLIST_INSTALLATION/public_html/simplesamlphp`
+- `cp -r config-templates config`
+- `cp -r metadata-templates metadata`
+
+* **In [PATH_TO_PHPLIST_INSTALLATION/public_html/simplesamlphp/config/authsources.php] the following parameters have to be set:**
+
+- **`entityID`**: The `entityID` is essentially the client ID which is specified in Keycloak or IDP
+- **`idp`**: The IDP is the indentifier for the IdP (Keycloak) which simplesaml would connect to.
+- **`RelayState`**: The `RelayState` specifies where `simplesamlphp` should redirect to after a successful authentication. Basically it's like a callback url. This should simply be the URL from which the authentication started. Hence, a 'redirect back'.
+- **`NameIDPolicy`**: The IdP is expected to return a `NameID` every successful auth session, this name ID is what identifies the user. Depending on the IdP this `NameID` might change every session. That makes it impossible to tract the user accross session. So we have to said the `NameIDPolicy` to `persistent` essentially telling the IdP to send thesame `NameID` all the time for the same user.
+
+The`authsources.php` should look like:
+
+```php
+<?php
+
+$config = [
+
+   //...
+   //...
+
+   // An authentication source which can authenticate against SAML 2.0 IdPs.
+   'default-sp' => [
+       'saml:SP',
+       // The entity ID of this SP.
+       // Can be NULL/unset, in which case an entity ID is generated based on the metadata URL.
+       'entityID' => 'account',
+       // The entity ID of the IdP this SP should contact.
+       // Can be NULL/unset, in which case the user will be shown a list of available IdPsnt.
+       'idp' => 'https://sso.phplist.com:8443/auth/realms/master',
```

Comment:
Preferably, this is configured in phpList and then applied here automatically.

Reply:
This is out of scope because the setting lives in `simplesamlphp` and not the plugin itself, to achieve this the plugin has to have code to modify `simplesamlphp` configuration files.